Add legend support for boxplots (#27840)
matplotlib/matplotlib: doc/users/next_whats_new/boxplot_legend_support.rst

Legend support for Boxplot
~~~~~~~~~~~~~~~~~~~~~~~~~~

> Boxplots now generate legend entries and can be labelled with the
> new *label* parameter. If a patch is passed to the box with ``patch_artist=True``,
> the legend gets its handle from the patch instead of the `.Line2D` object from
> the medians.

Reviewer: ```suggestion
Boxplots now support a *label* parameter to create legend entries.
```
The second sentence on the artists is a technical detail that is not relevant in this high-level overview. It's sufficient to have this in the docstring. But you could add a sentence on str vs list of str, because these enable to fundamentally different use cases of labelling.

Legend support for Boxplot
~~~~~~~~~~~~~~~~~~~~~~~~~~
Boxplots now support a *label* parameter to create legend entries.
Legend labels can be passed as a string to each individual boxplot
or as a list of strings to label multiple boxes in a single
boxplot call.

> .. plot::

Reviewer: I suggest to also add a plot for the single-string case. Something like https://github.com/matplotlib/matplotlib/pull/27840#discussion_r1518807844 - you may switch to `patch_artist=True` and color the boxes so that the distinction between the two sets is more obvious than just the color of the medians.

Author: Do we want this behaviour when boxplots with patches are mixed with ones without?

![2box](https://github.com/matplotlib/matplotlib/assets/9763064/91651909-68d0-4899-933b-1b10ab059403)

Reviewer: There’s no way around this. Tge individual boxplots do not know about each other and a non-patch_artist, boxplot can only have a line as legend entry, whereas the patch_artist boxplots should have the box. It’s the responsibility of the user to create a consistent plot.

Reviewer: Just wondering: Shouldn’t the color of the line for data A not be the orange of the median line?

Author: Yes it should! For some reason it only does if showbox=False. I didn't notice it before

Legend support for Boxplot
~~~~~~~~~~~~~~~~~~~~~~~~~~

> Boxplots now support a *label* parameter to create legend entries.

Reviewer 2: Should probably link to `Axes.boxplot` .
